src/python/pants/base/parse_context.py

```diff
@@ -58,7 +59,7 @@ def rel_path(self) -> str:
 
         :API: public
         """
-        return self._rel_path_oracle.rel_path()
+        return os.path.dirname(self._filepath_oracle.filepath())
```

Comment:
Uhm... it does still do what the docstring says. The `rel_path` was passed as `os.path.dirname(filepath)` before (in the call to `ParseState.reset()`) , so this is still exactly the same, only that we preserve the `filepath` through the `ParseState`, rather than just the `file_path`. Why is answered in the other comment on this topic.

Reply:
Ah, so the docstring was wrong even before (this is a relpath to a directory, but the docstring says it's a relpath to a BUILD file)?  If so, might be good to fix that docstring since you're here anyway.

Reply:
Oh, ops. Yeah in that case, heh. Didn't read the docstring too closely :P